presto-main/src/main/java/com/facebook/presto/execution/RevokeRolesTask.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2017, Teradata Corp. All rights reserved.
+ */
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.MetadataUtil;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.spi.security.PrestoPrincipal;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.RevokeRoles;
+import com.facebook.presto.transaction.TransactionManager;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static com.facebook.presto.metadata.MetadataUtil.createCatalogName;
+import static com.facebook.presto.metadata.MetadataUtil.createPrincipal;
+import static com.facebook.presto.spi.security.PrincipalType.ROLE;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MISSING_ROLE;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableSet;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class RevokeRolesTask
+        implements DataDefinitionTask<RevokeRoles>
+{
+    @Override
+    public String getName()
+    {
+        return "GRANT ROLE";
+    }
+
+    @Override
+    public CompletableFuture<?> execute(RevokeRoles statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    {
+        Session session = stateMachine.getSession();
+
+        Set<String> roles = statement.getRoles();
+        Set<PrestoPrincipal> grantees = statement.getGrantees().stream()
+                .map(MetadataUtil::createPrincipal)
+                .collect(toImmutableSet());
+        boolean adminOptionFor = statement.isAdminOptionFor();
+        Optional<PrestoPrincipal> grantor = statement.getGrantor().map(specification -> createPrincipal(session, specification));
+        String catalog = createCatalogName(session, statement, statement.getCatalog());
+
+        Set<String> availableRoles = metadata.listRoles(session, catalog);
```

Comment:
They you will either extract 3 lines, by cost of creation new class, or, if you want to extract the entire check code you need to introduce additional abstractions, like `RoleManipulationTask` abstract task that returns all the available roles.